### DISCUSSION TITLE
Thoughts on delegation graphs and weights

### DISCUSSION THREAD

**Discussion by davefrey** [liquidvotingio/decidim-module-liquidvoting#60] [Ideas]
Here's my mental model of delegation graphs and weights. 

![delegation graphs and weights](https://user-images.githubusercontent.com/29486/110612016-130fac00-8190-11eb-98f3-9ead3b75dddf.jpg)

When we talk about voting weights (to vote/support, delegate, or eventually abstain), there are direct and total weights; I assume the way our `votingResult` works is to reduce the graph to a **total weight**.

When considering presenting weights to users, here are some other things:

It would be nice if the current user were made aware of the weight(s), whether total or direct or both. If I were seriously interested in delegation, I might like to know both: direct is my personal influence, total is the whole weight of my vote/delegation/abstention.

It would be nice if the current user, if delegating, were made aware of the weight(s) (maybe only total is interesting here) for their delegate: how much influence does the delegate have?

This might also be a discovery attribute: do we present potential delegates with their weights (and maybe in context of the voting population size)?

It would also be nice if a delegating user knew the disposition of their delegation: has their delegate voted/delegated/abstained? And is that subject to change, or final (because close date, can't change one's vote, etc)?

**Comment by jinjagit** [comment]
Good summary. Thanks.

**Comment by oliverbarnes** [comment]
One thing to keep in mind is that the delegate's weight (and the current user's own weight) is transient up until a voting process ends, when all delegations and votes will have been cast.

Specially so if delegations are only proposal-focused, less so if the delegations are by category (unimplemented yet) or global and relatively more permanent. 

Another factor of uncertainty is whether delegators end up overriding their delegations with their own votes. 

So what they see when voting or delegating early on in a voting process can be very different from what they see when doing both at the end of it.

Hence I expect visualizing weights can be very misleading at times, and might incentivize for everybody to wait until the last minute to try to get a clearer picture... Which only adds to the uncertainty.

All assumptions, of course. Would be great to A/B test them

**Reply by davefrey** [reply]
Thanks for these details, so much to learn! "Blind" delegation sure simplifies things.

I hadn't thought about waiting to get a clearer picture. If we go down this weight transparency path, in addition to A/B testing, we'll want our telemetry to help us (and admins, users) understand churn or other ways to understand the lifecycle of a voting.

**Reply by oliverbarnes** [reply]
Yeah, blind delegation is probably the way to go initially, and down the line we can test out opening up some delegation info, see how it goes. 

[Analytics](https://gamedev.stackexchange.com/questions/115471/is-there-a-difference-between-analytics-and-telemetry) will definitely help with this. And true, we might be able to expose some patterns to users and admins after votings as well, it'll be very interesting to see those!
